include/cuco/detail/static_map/kernels.cuh

```diff
@@ -191,10 +225,88 @@ CUCO_KERNEL __launch_bounds__(BlockSize) void insert_or_apply_shmem(
     idx += loop_stride;
     while (idx < n) {
       value_type const& insert_pair = *(first + idx);
-      ref.insert_or_apply(insert_pair, op);
+      if constexpr (HasInit) {
+        ref.insert_or_apply(insert_pair, init, op);
+      } else {
+        ref.insert_or_apply(insert_pair, op);
+      }
       idx += loop_stride;
     }
   }
 }
 
+// TODO: Add docs
+template <bool HasInit,
+          int32_t CGSize,
+          typename Allocator,
+          typename InputIt,
+          typename Init,
+          typename Op,
+          typename Ref>
+void dispatch_insert_or_apply(
```

Comment:
This is not a kernel.

Reply:
moved the dispatch helper to `helpers.cuh` in https://github.com/NVIDIA/cuCollections/pull/555/commits/8791d655536beb053bfb4b9239ac273e336de321